src/config/ConfigBuilder.ts

```diff
@@ -383,6 +402,41 @@ export class ConfigBuilder {
         }
     }
 
+    /**
+     * Statistics properties are deprecated and metrics will be used instead. This method will ensure statistics properties
+     * and metrics config are not set at the same time. Also, it will set metricsConfig conditionally in four different cases:
+     *
+     * 1. When no config is given metrics will be enabled.
+     * 2. When only statistics props is given it will take effect.
+     * 3. When only metrics config is given it will take effect.
+     * 4. When both statistics props and metrics config are given throw.
+     */
+    private static handleMetricsAndStatistics(config: ClientConfig, metricsConfig: MetricsConfig): void {
+        if (config.hasOwnProperty('properties') && config.properties.hasOwnProperty('hazelcast.client.statistics.enabled')) {
```

Comment:
this will run without an error in a configuration like below
```js
{metrics: {collectionFrequencySeconds: 4}, properties: {'hazelcast.client.statistics.enabled': true}}
```

Reply:
yeah, but I think they don't create a conflict. I think we should allow that.

Reply:
Personally, I don't think this is valid usage. I don't think it should be possible to mix two configuration options. @yuce WDYT?

Reply:
How does the Java client behave in this case?

Reply:
```java
public static void main(String[] args) throws InterruptedException {
        ClientConfig c = new ClientConfig();
        c.getMetricsConfig().setCollectionFrequencySeconds(4);
        c.getProperties().setProperty("hazelcast.client.statistics.enabled", "true");
        HazelcastInstance h = HazelcastClient.newHazelcastClient(c);
        IMap<String, String> map = h.getMap("map");
        map.put("key", "value");
    }
```

results in 

Feb 10, 2022 1:07:03 PM com.hazelcast.client.impl.statistics.ClientStatisticsService
INFO: Client statistics is enabled with period 4 seconds.

with hazelcast commit b27dee57b8dca6e70071656e59b2cd8af100b1cb (9th Feb)

Reply:
```java
public static void main(String[] args) throws InterruptedException {
        ClientConfig c = new ClientConfig();
        c.getMetricsConfig().setCollectionFrequencySeconds(4);
        c.setProperty("hazelcast.client.statistics.enabled", "false");
        HazelcastInstance h = HazelcastClient.newHazelcastClient(c);
        IMap<String, String> map = h.getMap("map");
        map.put("key", "value");
    }
```

with this code, statistics is disabled.

Reply:
Java does not throw in case both are set: 

```java
public static void main(String[] args) throws InterruptedException {
        ClientConfig c = new ClientConfig();
        c.setProperty("hazelcast.client.statistics.enabled", "false");
        c.getMetricsConfig().setEnabled(true);

        HazelcastInstance h = HazelcastClient.newHazelcastClient(c);
        IMap<String, String> map = h.getMap("map");
        map.put("key", "value");
    }
```

It gived precedence to statistics property and disables statistics. Should we do the same @yuce @mdumandag ?

Reply:
Hmm, I think it is good to be consistent with Java client then

Reply:
done! @yuce stale review dismissed 😉